src/main/java/io/codelex/flightplanner/flight/FlightDatabaseService.java

```diff
@@ -0,0 +1,148 @@
+package io.codelex.flightplanner.flight;
+
+import io.codelex.flightplanner.airport.AirportService;
+import io.codelex.flightplanner.airport.domain.Airport;
+import io.codelex.flightplanner.error.*;
+import io.codelex.flightplanner.flight.domain.Flight;
+import io.codelex.flightplanner.flight.request.AddFlightRequest;
+import io.codelex.flightplanner.flight.request.SearchFlightRequest;
+import io.codelex.flightplanner.flight.response.AddFlightResponse;
+import io.codelex.flightplanner.flight.response.GetFlightResponse;
+import io.codelex.flightplanner.flight.response.SearchFlightResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+
+public class FlightDatabaseService implements FlightService {
+
+    private final FlightRepository flightRepository;
+    private final AirportService airportService;
+    Logger logger = LoggerFactory.getLogger(FlightDatabaseService.class);
+
+    public FlightDatabaseService(FlightRepository flightRepository, AirportService airportService) {
+        this.flightRepository = flightRepository;
+        this.airportService = airportService;
+    }
+
+    @Override
+    public synchronized AddFlightResponse addFlight(AddFlightRequest flightRequest) {
+        logger.debug("Got request: " + flightRequest);
+        if (validateAddFlightRequest(flightRequest)) {
+
+            Airport fromAirport = airportService.addAirport(flightRequest.getFrom());
+            Airport toAirport = airportService.addAirport(flightRequest.getTo());
+
+            Flight newFlight = new Flight(fromAirport, toAirport, flightRequest.getCarrier(), flightRequest.getDepartureTime(), flightRequest.getArrivalTime());
+
+            Flight addedFlight = flightRepository.save(newFlight);
+            logger.info("Added flight from service to db Flight: " + newFlight);
+            return new AddFlightResponse(addedFlight);
+        } else {
+            throw new AddFlightException("Failed to add new Flight with request: " + flightRequest);
+        }
+    }
+
+
+    @Override
+    public boolean validateAddFlightRequest(AddFlightRequest flightRequest) {
+        boolean flightDoesntExist = validateFlight(flightRequest);
+        boolean airportsAreValid = validateRequestAirports(flightRequest);
+        boolean datesAreValid = validateDates(flightRequest);
+
+        return (airportsAreValid && datesAreValid && flightDoesntExist);
+    }
+
+    @Override
+    public boolean validateFlight(AddFlightRequest addFlightRequest) {
+
+        for (Flight flight : flightRepository.findFlightByArrivalTimeAndDepartureTime(
```

Comment:
Te drīzāk uzreiz datubāzē meklē pēc arrivalTime, departure time un lidostām - datubāze to vienkārš izdarīs ātrāk.

Reply:
Papildināju meklēšanu, lai meklē arī pēc 'to' un 'from' Airportiem